sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQueryManagerSuite.scala

```diff
@@ -239,6 +237,40 @@ class StreamingQueryManagerSuite extends StreamTest with BeforeAndAfter {
     }
   }
 
+  test("stopAllQueries") {
+    val datasets = Seq.fill(5)(makeDataset._2)
+    withQueriesOn(datasets: _*) { queries =>
+      assert(queries.forall(_.isActive))
+      spark.streams.stopAllQueries()
+      assert(queries.forall(_.isActive == false), "Queries are still running")
+    }
+  }
+
+  test("stop session stops all queries") {
+    val inputData = MemoryStream[Int]
+    val mapped = inputData.toDS.map(6 / _)
+    var query: StreamingQuery = null
+    try {
+      query = mapped.toDF.writeStream
+        .format("memory")
+        .queryName(s"queryInNewSession")
+        .outputMode("append")
+        .start()
+      assert(query.isActive)
+      spark.stop()
+      assert(spark.sparkContext.isStopped)
+      assert(query.isActive == false, "Query is still running")
+    } catch {
+      case NonFatal(e) =>
+        if (query != null) query.stop()
+        throw e
```

Comment:
why still try/catch to stop the query? since this is a test of this specific behavior, if the query isn't stopped, or throws, the test actually is failing?

more importantly, why ignore NonFatal exception thrown?